### PR TITLE
Fix FOMO head

### PR DESCRIPTION
**Pull request by JSabadin**
# FOMO Model Updates and Improvements

The **FOMO model**, with its very simple architecture, can sometimes cause issues in object detection tasks. These issues have been addressed in this pull request.

## Identified Issue: Multiple Predictions Per Object

In its default configuration, the model occasionally predicts multiple points for each object. This behavior can be mitigated by:

1. **Reducing the Loss Weight:** Lowering the weight applied to object centers in the loss function significantly reduces the likelihood of multiple predictions.
2. **Applying Non-Maximum Suppression (NMS):** If adjusting the loss weight does not fully resolve the issue, the model includes an optional, simple, and fast NMS (Non-Maximum Suppression) step to filter redundant predictions.

## Demonstration of Results

The following images illustrate the model's performance under different configurations:

1. **Before Adjustment (Loss Weight: 1000, No NMS):**  
   The model predicts multiple points for each object.

   Before Adjustment

2. **After Adjustment 1 (Loss Weight: 500, With NMS):**  
   The model's predictions are improved with fewer redundant points, thanks to a combination of reduced loss weight and NMS.

   After Adjustment 1

3. **After Adjustment 2 (Loss Weight: 5, No NMS):**  
   A significantly lower loss weight further simplifies the predictions, even without applying NMS.

   After Adjustment 2
   
 ## Validation Loss for Larger Objects

One remaining issue with this simple model is that the validation loss can rise for larger objects. This occurs because, in the initial training stages, the model tends to make uncertain and excessive predictions. For example:

- **Initial State:** The model predicts numerous uncertain points for each object.  
  Initial Predictions

- **After a Few Epochs:** The model gradually learns to identify the centers of objects but can still occasionally misplace them due to the inherent difficulty in defining the "correct" center in the labels.  
  Refined Predictions

This behavior is important to keep in mind when using this model, especially for datasets with larger or irregularly shaped objects.

---

### CODE REVIEW

File: luxonis_train/nodes/heads/fomo_head.py [hunk 1]
```diff
@@ -14,12 +15,13 @@
 class FOMOHead(BaseNode[list[Tensor], list[Tensor]]):
     tasks: list[TaskType] = [TaskType.KEYPOINTS, TaskType.BOUNDINGBOX]
     in_channels: int
-    attach_index: int = 2
+    attach_index: int = 1
```
[the remark this answers is not in the document]
Comment (JSabadin) [reply luxonis_train/nodes/heads/fomo_head.py:18]: This was designed to match the feature dimensions used in the original article.

**Review by klemen1999 — APPROVED** [review]

**Review by kozlov721 — APPROVED** [review]
